apis/vpcaccess/v1beta1/connector_reference.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/controller/direct"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ refsv1beta1.ExternalNormalizer = &VPCAccessConnectorRef{}
+
+// VPCAccessConnectorRef defines the resource reference to VPCAccessConnector, which "External" field
+// holds the GCP identifier for the KRM object.
+type VPCAccessConnectorRef struct {
+	// A reference to an externally managed VPCAccessConnector resource.
+	// Should be in the format "projects/<projectID>/locations/<location>/connectors/<connectorID>".
+	External string `json:"external,omitempty"`
+
+	// The name of a VPCAccessConnector resource.
+	Name string `json:"name,omitempty"`
+
+	// The namespace of a VPCAccessConnector resource.
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// VPCAccessConnectorID defines the unique identifier for a VPCAccessConnector.
+type VPCAccessConnectorID struct {
+	// Immutable.
+	// The VPCAccessConnector name. If not given, the metadata.name will be used.
+	ResourceID *string `json:"resourceID,omitempty"`
```

Comment:
Suggest changing the `ResourceID` from a pointer to a val, and make it private, so the code gives a clear signal that this field is not supposed to be changed during reconciliation (unless for service-generated ID resource).

Reply:
never mind, somehow I misread the code ....